src/sherpa_ai/tools.py

```diff
@@ -83,13 +93,36 @@ class SearchTool(BaseTool):
         "Access the internet to search for the information. Only use this tool when "
         "you cannot find the information using internal search."
     )
-
-    def augment_query(self, query) -> str:
-        return query + " site:" + self.config.gsite if self.config.gsite else query
-
+    
     def _run(self, query: str) -> str:
-        query = self.augment_query(query)
+        result = ""
+        if self.config.gsite:
+            gsite_list = self.config.gsite.split(", ")
+            gsite_list = [i for i in gsite_list if i != " " and i != "\n" and i != None]
+            if False in [validate_url(i) for i in gsite_list]:
```

Comment:
This is a bit duplicated as we are doing this check for every call of the search. At this point, the URL in the configuration should already be valid. I think we should put this check inside the AgentConfig. We can create a new attribute in the AgentConfig called `search_domains` as a list. Then, we should parse the `gsite` string into list of URLs in [this method](https://github.com/Aggregate-Intellect/sherpa/blob/93677ee82f250a3248668b8fbc318393583b8cf0/src/sherpa_ai/config/task_config.py#L29). Then we don't need to repeat this check here.